nixos/modules/hardware/video/nvidia.nix

```diff
@@ -285,7 +285,7 @@ in
             KERNEL=="nvidia_uvm", RUN+="${pkgs.runtimeShell} -c 'mknod -m 666 /dev/nvidia-uvm-tools c $$(grep nvidia-uvm /proc/devices | cut -d \  -f 1) 1'"
           '';
           hardware.opengl = {
-            extraPackages = [ nvidia_x11.out nvidia_x11.settings.libXNVCtrl ];
+            extraPackages = [ nvidia_x11.out ] ++ (lib.optional (builtins.hasAttr "libXNVCtrl" nvidia_x11.settings) nvidia_x11.settings.libXNVCtrl);
```

Comment:
This seems fairly unpredictable, libXNVCtrl being included in the impure deployment depending on nixpkgs overlays rather than explicit nixos attributes. In future, I think a better approach would be to

0) Verify if the library is actually driver-locked and can't be linked via RUNPATHs
1) Introduce a nixos option for "mount libxnvctrl"; the path should be consistent with  whatever resolution is eventually found in the issue about renaming "hardware.opengl" into "hardware.drivers"
2) Option being enabled, we should add an assertion that the passthru attribute exist; the message should say something like "hardware.nvidia.package doesn't expose libXNVCtrl; pick a package that exposes the attribute"